src/main/cpp/controllers/DrivetrainController.cpp

```diff
@@ -73,6 +73,16 @@ void DrivetrainController::AbortTrajectories() {
     m_trajectory = frc::Trajectory{};
 }
 
+bool DrivetrainController::AtHeading() const {
+    units::radian_t ref{m_r(State::kHeading)};
+    return (ref >= m_headingGoal + 0.02_rad) &&
+           (ref <= m_headingGoal - 0.02_rad);
+}
+
+bool DrivetrainController::HaveHeadingGoal() const {
+    return m_headingGoal > 0_rad;
```

Comment:
The heading goal could still be valid and have a negative value, so this check is incorrect.

If m_headingGoal can be invalid at any point, it would make more sense to make the heading controller a completely separate class from DrivetrainController that the Drivetrain subsystem switches between. This promotes separation of responsibilities and makes it easier to debug.